machine-learning/src/main.ts

```diff
@@ -7,7 +7,7 @@ async function bootstrap() {
 
   const port = Number(process.env.MACHINE_LEARNING_PORT) || 3003;
 
-  await app.listen(port, () => {
+  await app.listen(port, '::', () => {
```

Comment:
I don't think this is necessary, it should listen to ipv6 automatically, if it's available.

https://nodejs.org/api/net.html#serverlistenport-host-backlog-callback

> If host is omitted, the server will accept connections on the [unspecified IPv6 address](https://en.wikipedia.org/wiki/IPv6_address#Unspecified_address) (::) when IPv6 is available, or the [unspecified IPv4 address](https://en.wikipedia.org/wiki/0.0.0.0) (0.0.0.0) otherwise.